src/EnergyPlus/WaterUse.cc

```diff
@@ -1092,117 +1099,142 @@ namespace WaterUse {
         this->TotalMassFlowRate = this->TotalVolFlowRate * Psychrometrics::RhoH2O(DataGlobalConstants::InitConvTemp);
 
         // Calculate hot and cold water mixing at the tap
-        if (this->TotalMassFlowRate > 0.0) {
+        if (this->TotalMassFlowRate > 0.0 && this->allowHotControl) {
             // Calculate the flow rates needed to meet the target temperature
-            if (this->HotTemp == this->ColdTemp) { // Avoid divide by zero
-                // There is no hot water
+            if (this->TargetTemp <= this->ColdTemp + EPSILON) {
+                // don't need to mix (use cold) or hot water flow would be very small if within EPSILON above cold temp (use cold)
```

Comment:
Here is the tier if block. What to do. It's a logical method using the target temp. If too cold, use cold. If too hot, use hot. If in between, use most relavent logic.

Reply:
One more thing is the use of EPSILON here and other places should? mean that the final calculation will never be < 0 or > max. Fingers crossed.

    this->HotMassFlowRate = this->TotalMassFlowRate * (this->TargetTemp - this->ColdTemp) / (this->HotTemp - this->ColdTemp);